models/article-model.js

```diff
@@ -30,6 +30,27 @@ GROUP BY
     articles.article_id
 ORDER BY 
     articles.created_at DESC`
-      )
-      .then(({ rows }) => rows);
-  };
\ No newline at end of file
+    )
+    .then(({ rows }) => rows);
+};
+
+exports.updateArticleById = (article_id, inc_votes) => {
+  const articleExistsQuery =
+    "SELECT EXISTS(SELECT 1 FROM articles WHERE article_id = $1);";
+  const articleUpdateQuery = `
+      UPDATE articles
+      SET votes = votes + $1
+      WHERE article_id = $2
+      RETURNING *
+    `;
+  return db
+    .query(articleExistsQuery, [article_id])
+    .then(({ rows }) => {
+      if (!rows[0].exists) {
+        return Promise.reject({ status: 404, msg: "not found" });
+      }
+
+      return db.query(articleUpdateQuery, [inc_votes, article_id]);

```

Comment:
We should be able to do this as one db query. Consider a check exists function that would be resuable. You may need a promise all to achieve it.